### PR TITLE
Add keyboard interactor

### LINKED ISSUE #12: Keyboard emulation for interactors
I am spinning this off from thefrontside/bigtest#818 as its own issue, since while it is related, this feature also stands on its own.

When writing tests for keyboard interactions, we are often forced by test frameworks to write them in an unnatural way. The keyboard is a physical device which the user can use at any point in time. Pressing keys on the keyboard can have various effects on the document, depending on which element currently has focus, which keys are pressed, and which event handlers are attached.

Under the hood, keyboard events such as `keydown` and `keyup` are fired for the currently focused element. This is the level that testing libraries usually attach themselves too, but it is worth considering that this is *not* how a user sees it. Sometimes the user will deliberately move focus to a certain element before pressing a key, other times, they will be unaware of where focus currently resides.

Since the keyboard is a distinct physical device, and that is how the user sees it, and in BigTest we are trying to model the user's behaviour, why not model the keyboard as such?

### How would this work?

The keyboard becomes a global interactor, which can be used both on its own, and also in part of composed actions. My preferred design is to make keyboard interactions part of the `Page` interactor. It already exists and models the document in its entirety.

``` typescript
Page.pressKey('a', { meta: true });
```

This would sent the key `a` to the currently focused element.

We could add a helper for typing longer strings of text, so we don't have to sent each individual keystroke:

``` typescript
Page.typeText('Hello world');
```

This helper could have a simple DSL for including control characters:

``` typescript
Page.typeText('Hello world<enter>');
```

We could even have a low level API for holding and releasing keys for even more complex interactions:

``` typescript
Page.holdKey('meta');
Page.holdKey('a');
Page.releaseKey('meta');
Page.releaseKey('a');
```

### Keyboard layouts

A future extension to this feature could be support for various keyboard layouts. A keyboard layout is really a mapping of physical keys to expected characters, and vice versa. From a user's perspective they would normally be hitting a key with a specific character, but there may even be rare occasions where they would hit a key in a specific location, rather than with an expected value.

For example, imagine a game where the `WASD` keys are used to control movement. Using a qwerty keyboard layout, this is natural and convenient, but on a dvorak keyboard, it would be massively inconvenient. On dvorak we would want to use the keys labeled `<AOE` for this purpose, since they are in the same location as the `WASD` keys on dvorak. So in this case, the key *location* is more important than the key *value*. It is worth noting that this is a rare case though, and while we should probably *support* this case, it is not the case we should *optimize* for.

### PR DESCRIPTION
**Pull request by jnicklas**
Closes #12

The keyboard interactor can send both individual keypresses, and type string of text. It always operators on the currently focused/active element.

Also adds a global KeyboardLayout. The layout handles mapping of physical keys to key values.

### CODE REVIEW

File: packages/html/src/fill-in.ts [hunk 1]
```diff
@@ -60,17 +21,18 @@ function setValue(element: TextFieldElement, value: string): void {
...
+  await clear(interactor);
+  await Keyboard.type(value);
 
   if(originalValue !== value) {
-    dispatchChange(element);
+    await interactor.perform(dispatchChange);
   }
 
-  element.blur();
+  await interactor.perform((element) => element.blur());
```
Comment (wKich) [inline packages/html/src/fill-in.ts:37]: The whole fillIn action must be synchronous unless the second interaction action after focus or clear will fail in some circumstances, like if you use `TextField({ value: 'Foo', focused: false }).fillIn('Bar')`

File: packages/globals/src/index.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export * from './globals'
+export { KeyboardLayout, KeyCode, US_INTERNATIONAL } from './keyboard-layout'
```
Comment (wKich) [inline packages/globals/src/index.ts:2]: Maybe it's better to move it in a separate package `@interactors/keyboard`?

File: packages/html/src/index.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ export { Interaction, ReadonlyInteraction, isInteraction } from './interaction';
 export { createInteractor } from './create-interactor';
 export { createInspector } from './inspector'
 export { Page } from './page';
+export { Keyboard } from './keyboard';
```
Comment (wKich) [inline packages/html/src/index.ts:6]: Should we move it to a separate package?
```suggestion
export { Keyboard } from '@interactors/keyboard';
```

File: packages/html/src/keyboard.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        }
+        // keyup is always dispatched
+        dispatchKeyUp(activeElement, options);
+      });
+    },
+
+    async type(interactor, value: string) {
+      for(let letter of value) {
+        await Keyboard.press({ key: letter });
+      }
```
Comment (wKich) [inline packages/html/src/keyboard.ts:41]: What about using `userEvent.type`?
https://github.com/testing-library/user-event/blob/main/src/type/index.ts
https://testing-library.com/docs/ecosystem-user-event/#typeelement-text-options

File: packages/html/test.txt [hunk 5]
```diff
@@ -0,0 +1,103 @@
+"Backspace", "Backspace"
```
Comment (wKich) [inline packages/html/test.txt:1]: Do we need this file?

**Review by cowboyd — DISMISSED** [review]

**Comment by wKich** [comment]
@cowboyd why did you approve it? There is definitely a bug with fillIn function

File: packages/html/test.txt (hunk 5 above)
Comment (jnicklas) [reply packages/html/test.txt:1]: No, it was accidentally committed, I'll remove it.

File: packages/globals/src/index.ts (hunk 2 above)
Comment (cowboyd) [reply packages/globals/src/index.ts:2]: I think that this is core to `html` as this keyboard interactor is very much tied to the DOM. We could move it to another package, but `fillIn` currently needs it, so we'd have a circular dependency without a lot of work.

**Review by cowboyd — APPROVED** [review]

**Pull request merged**